env0/resource_project_policy.go

```diff
@@ -0,0 +1,174 @@
+	d.Set("include_cost_estimation", policy.IncludeCostEstimation)
+	d.Set("skip_apply_when_plan_is_empty", policy.SkipApplyWhenPlanIsEmpty)
+	d.Set("disable_destroy_environments", policy.DisableDestroyEnvironments)
+	d.Set("skip_redundant_deployments", policy.SkipRedundantDepolyments)
+	d.Set("updated_by", policy.UpdatedBy)
+}
+
+func resourcePolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourcePolicyUpdate(ctx, d, meta)
+}
+
+func resourcePolicyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	policy, err := apiClient.Policy(d.Id())
+	if err != nil {
+		return diag.Errorf("could not get policy: %v", err)
+	}
+
+	setPolicySchema(d, policy)
+
+	return nil
+}
+
+func resourcePolicyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	payload := client.PolicyUpdatePayload{}
+	if projectID, ok := d.GetOk("project_id"); ok {
+		payload.ProjectId = projectID.(string)
+		log.Printf("[INFO] updating policy for project: %s\n", projectID)
+	} else {
+		return diag.Errorf("project id not ok")
+	}
+	if numberOfEnvironments, ok := d.GetOk("number_of_environments"); ok {
+		payload.NumberOfEnvironments = numberOfEnvironments.(int)
+	}
+	if numberOfEnvironmentsTotal, ok := d.GetOk("number_of_environments_total"); ok {
+		payload.NumberOfEnvironmentsTotal = numberOfEnvironmentsTotal.(int)
+	}
+	if requiresApprovalDefault, ok := d.GetOk("requires_approval_default"); ok {
+		payload.RequiresApprovalDefault = requiresApprovalDefault.(bool)
+	}
+	if includeCostEstimation, ok := d.GetOk("include_cost_estimation"); ok {
+		payload.IncludeCostEstimation = includeCostEstimation.(bool)
+	}
+	if skipApplyWhenPlanIsEmpty, ok := d.GetOk("skip_apply_when_plan_is_empty"); ok {
+		payload.SkipApplyWhenPlanIsEmpty = skipApplyWhenPlanIsEmpty.(bool)
+	}
+	if disableDestroyEnvironments, ok := d.GetOk("disable_destroy_environments"); ok {
+		payload.DisableDestroyEnvironments = disableDestroyEnvironments.(bool)
+	}
+	if skipRedundantDeployments, ok := d.GetOk("skip_redundant_depolyments"); ok {
+		payload.SkipRedundantDepolyments = skipRedundantDeployments.(bool)
+	}
+
+	_, err := apiClient.PolicyUpdate(payload)
+	if err != nil {
+		return diag.Errorf("could not update policy: %v", err)
+	}
+	d.SetId(payload.ProjectId)
+	return nil
+}
+
+func resourcePolicyReset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	id := d.Id()
+
+	payload := client.PolicyUpdatePayload{
+		ProjectId: id,
+	}
+
+	_, err := apiClient.PolicyUpdate(payload)
+	if err != nil {
+		return diag.Errorf("could not delete policy: %v", err)
+	}
+
+	return nil
+}
+
+func resourcePolicyImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
1. One should be able to import a policy by the `project id` as well.
2. I think you're missing a `d.setId()` call here. IIRC otherwise terraform wouldn't save it to state. Correct me if I'm wrong please.

Reply:
`id` is the `project_id`. I have updated the naming to reflect that this. That is the only way to look up a project policy, according to the API docs.

The `terraform.tfstate.backup` file saved after running this integration test indicates that the objects are being saved correctly. The `id` is set in the state, so it does not need to be set again.

Reply:
`id` is the `project_id`. I have updated the naming to reflect that this. That is the only way to look up a project policy, according to the API docs.

The `terraform.tfstate.backup` file saved after running this integration test indicates that the objects are being saved correctly. The `id` is set in the state, so it does not need to be set again.